connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteThreadRunner.scala

```diff
@@ -238,17 +238,45 @@ private[connect] class ExecuteThreadRunner(executeHolder: ExecuteHolder) extends
         }
         completed = true // no longer interruptible
 
-        if (executeHolder.reattachable) {
-          // Reattachable execution sends a ResultComplete at the end of the stream
-          // to signal that there isn't more coming.
-          executeHolder.responseObserver.onNextComplete(createResultComplete())
-        } else {
-          executeHolder.responseObserver.onCompleted()
+        // If the request returns a long running iterator (e.g. StreamingQueryListener needs
+        // a long-running iterator to continuously stream back events),
+        // we delegate the sending of the final ResultComplete to the handler itself.
```

Comment:
it's not only the final ResultComplete, it's also all the events that are sent from the listener thread?
Maybe mention this in the comment here, that the execution thread exits, but more replies can be sent from somewhere else (background?) and then the result complete is also delegated there.